src/ClientData/include/ClientData/TimerData.h

```diff
@@ -29,6 +29,12 @@ class TimerData final : public TimerDataInterface {
   [[nodiscard]] std::vector<const orbit_client_protos::TimerInfo*> GetTimers(
       uint64_t min_tick = std::numeric_limits<uint64_t>::min(),
       uint64_t max_tick = std::numeric_limits<uint64_t>::max()) const override;
+  [[nodiscard]] std::vector<const orbit_client_protos::TimerInfo*> GetTimersAtDepthDiscretized(
+      uint32_t /*depth*/, uint32_t /*resolution*/, uint64_t /*start_ns*/,
+      uint64_t /*end_ns*/) const override {
+    return {};
```

Comment:
What is the semantic implication of that function returning an empty vector here?

As far as I can see, you are going to call the method on the interface in order to get all the timer infos to be drawn. So in case of TimerData, this function should be implemented at some point, right?

If so, I would add an ORBIT_UNREACHABLE() with a TODO refering to the bug to implement that optimisation. Or even better:

Return GetTimers for now, so the function is still valid but not yet optimized. 

WDYT?

Reply:
I will implement it in a follow-up PR. I'm working on improving TimerDataTest first but I have a local implementation that will come in a following PR. But I agree, ORBIT_UNREACHABLE makes sense for now.